File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        char[] input = br.readLine().toCharArray();
        int len = input.length;
        int cur = 0;
        int idx = 0;
        
        outer: while(true) {
            cur++;
            char[] cur_list = Integer.toString(cur).toCharArray();
            for(char c : cur_list) {
                if(c == input[idx]) {
                    idx++;
                    if(idx >= len) {
                        System.out.println(cur);
                        break outer;
                    }
                }
            }
        }
    }
}

File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.swift
let input = readLine()!.map{Int(String($0))!}
var cur = 0
var idx = 0

outer: while(true) {
    cur += 1
    let list = String(cur).map{Int(String($0))!}
    for s in list {
        if(input[idx] == s) { 
            idx += 1
            if(idx >= input.count) {
                print(cur)
                break outer
            }
        }
    }
}